### PR TITLE
support singel build and multiarch build

### PR DESCRIPTION
**Pull request by MouceL**

### CODE REVIEW

File: bin/multiarch.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+# usage: create multiarch image
+#   PREFIX="docker.io/slimeio/slime-limiter:" ./multiarch.sh --push v1 \
+#   v1_linux_amd64 v1_linux_arm64
+##   or
+#   ./multiarch.sh --push docker.io/slimeio/slime-limiter:v1 \
+#   slimeio/slime-limiter:v1_linux_arm64-dirty \
+#   slimeio/slime-limiter:v1_linux_amd64-dirty
+
+# usage: deploy multiarch image for pilot, tools and etc
```
Comment (YonkaFang) [inline bin/multiarch.sh:11]: correct it

File: bin/multiarch.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+## will to build,image,push,manifest-create,manifest-push for specified archs and target
+#  ./multiarch.sh deploy arm64 amd64
+
+set -o errexit
+# set -x
+
+if [[ "$1" == "deploy" ]]; then
```
Comment (YonkaFang) [inline bin/multiarch.sh:18]: correct it

File: bin/publish.sh [hunk 3]
```diff
@@ -1,56 +1,56 @@
...
+  else
+    TARGET_GOOS=linux
+  fi
+fi
 
 if test -z "$MOD"; then
   fatal "empty MOD"
 fi
 
-function calc_unstaged_hash() {
```
Comment (YonkaFang) [inline bin/publish.sh:17]: why remove this?

File: bin/publish.sh [hunk 4]
```diff
@@ -1,56 +1,56 @@
 #!/usr/bin/env bash
 
-HUB=${HUB:-"docker.io/slimeio registry.cn-hangzhou.aliyuncs.com/slimeio"}
```
Comment (YonkaFang) [inline bin/publish.sh:3]: support of `PUSH_HUBS` is lost?

**Comment by MouceL** [comment]
image

File: bin/publish.sh [hunk 5]
```diff
@@ -22,35 +24,59 @@ function calc_unstaged_hash() {
...
-commit=$(git rev-parse --short HEAD)
-if [[ -z "${version}" ]]; then
-  tag=$(git show-ref --tags| grep "$commit" | awk -F"[/]" '{print $3}'|tail -1)
-  if [[ -z "${tag}" ]]; then
-    branch=$(git symbolic-ref --short -q HEAD)
-    if [[ -n "${branch}" ]]; then
-      version=$branch
+
+
+if [[ -z "$TAG" ]]; then
```
Comment (YonkaFang) [inline bin/publish.sh:50]: the `get-tag` logic in origin slime publish.sh is more accurate

File: bin/publish.sh [hunk 6]
```diff
@@ -62,51 +88,36 @@ else
...
-    # rest param will be consider as info to print, like: *.sh print image
-    set +x
-    shift
-    print_info "$@"
-    break
-    ;;
-  push)
-    docker push "${image_url}"
+    echo "build docker image: ${image}" >&2
+    if [[ "$TARGET_GOOS" == "linux" && "$TARGET_GOARCH" == "amd64" && ("$(uname -p)" == "x86_64" || "$(uname -p)" == "i386") ]]; then
```
Comment (YonkaFang) [inline bin/publish.sh:99]: I found that this condition-branch is meaningless, we can remove it and use `docker buildx build` line directly

File: bin/publish.sh [hunk 7]
```diff
@@ -22,35 +24,58 @@ function calc_unstaged_hash() {
...
 if [[ -z "${IGNORE_DIRTY}" && -n "$(git status -s --porcelain)" ]]; then
   unstaged_hash=$(calc_unstaged_hash)
   dirty="-dirty_${unstaged_hash::7}"
 fi
-commit=$(git rev-parse --short HEAD)
-if [[ -z "${version}" ]]; then
+
+if [[ -z "$TAG" ]]; then
+  branch=$(git symbolic-ref --short -q HEAD)
   tag=$(git show-ref --tags| grep "$commit" | awk -F"[/]" '{print $3}'|tail -1)
```
Comment (YonkaFang) [inline bin/publish.sh:51]: we can not use `commit` until we get it

**Pull request merged**